packages/vsce/src/commands/filterResourceCommands.ts

```diff
@@ -22,10 +22,12 @@ export function getFilterResourcesCommand(tree: CICSTree, treeview: TreeView<ICI
     if (!pattern) {
       return;
     }
-
-    await node.getChildResource().meta.appendCriteriaHistory(pattern);
-    node.setFilter([pattern]);
-    node.description = pattern;
+    //pattern is coming as "A,B,C,D..." so  need to split it into an array"
+    const patternArray = pattern.split(",").map((s) => s.trim());
+    const patternExpression = patternArray.join(" OR ");
+    await node.getChildResource().meta.appendCriteriaHistory(patternExpression);
+    node.setFilter(patternArray);
+    node.description = patternExpression;
```

Comment:
Could then to this to get the "PROGRAM=A OR PROGRAM=B" string, which will be more accurate than rebuilding it ourselves.
]
```suggestion
    node.description = node.getChildResource().resources.getFilter();
```